src/main.rs

```diff
@@ -747,34 +745,30 @@ const APP: () = {
     }
 
     #[task(binds=DMA1_STR3, resources=[adcs, dacs, iir_state, iir_ch], priority=2)]
-    fn adc_update(c: adc_update::Context) {
-        let (adc0_samples, adc1_samples) =
-            c.resources.adcs.transfer_complete_handler();
-
-        let (dac0, dac1) = c.resources.dacs.prepare_data();
-
-        for (i, (adc0, adc1)) in
-            adc0_samples.iter().zip(adc1_samples.iter()).enumerate()
-        {
-            dac0[i] = {
-                let x0 = f32::from(*adc0 as i16);
-                let y0 = c.resources.iir_ch[0]
-                    .update(&mut c.resources.iir_state[0], x0);
-                y0 as i16 as u16 ^ 0x8000
-            };
-
-            dac1[i] = {
-                let x1 = f32::from(*adc1 as i16);
-                let y1 = c.resources.iir_ch[1]
-                    .update(&mut c.resources.iir_state[1], x1);
-                y1 as i16 as u16 ^ 0x8000
-            };
+    fn process(c: process::Context) {
+        let adc_samples = [
+            c.resources.adcs.0.acquire_buffer(),
+            c.resources.adcs.1.acquire_buffer(),
+        ];
+        let dac_samples = [
+            c.resources.dacs.0.acquire_buffer(),
+            c.resources.dacs.1.acquire_buffer(),
+        ];
+
+        for channel in 0..adc_samples.len() {
+            for sample in 0..adc_samples[0].len() {
```

Comment:
should this be `0..adc_samples[channel].len()`? I don't think it should make a difference, but would make more sense logically.

Reply:
It doesn't matter since arrays are to be homogeneous. Let's leave it.